code/game/objects/items/bodybag.dm

```diff
@@ -84,6 +84,33 @@
 	else
 		icon_state = icon_opened
 
+/obj/structure/closet/body_bag/take_contents()
+	..()
+	var/mob/living/L = usr
+	for(L in src)
+		if (L.resting == FALSE) //if you are standing up and the body bag closes...
+			L.lay_down() //rest
+
+/obj/structure/closet/body_bag/close()
+	..()
+	if(usr)
+		var/mob/living/L = usr
+		if(L.loc == src) //only process bags when there's a mob inside
+			if(!processing_objects.Find(src))
+				processing_objects.Add(src)
+
+/obj/structure/closet/body_bag/open()
+	if(processing_objects.Find(src))
+		processing_objects.Remove(src) //no need to process bags when nothing is inside
+	..()
+
+
+/obj/structure/closet/body_bag/process()
```

Comment:
instead of looping through the mobs inside a bag every tick, it would make more since to create some proc that gets called when rest status is changed from the verb -- that proc could be used for other things too, like updating the icon of inflatable shelters, etc.